### ISSUE TITLE
Support for absolute path links (without hostname?)

### ISSUE THREAD

**Issue by zeckalpha** [pezra/hal-client#32]
From the hal spec:
~~~
   {
     "_links": {
       "self": { "href": "/orders/523" },
       "warehouse": { "href": "/warehouse/56" },
       "invoice": { "href": "/invoices/873" }
     },
     "currency": "USD",
     "status": "shipped",
     "total": 10.20
   }
~~~
On similar representations, I'm getting unknown scheme errors when following links.

**Comment by pezra** [comment]
It would be *great* to have support for relative links.  I think we could use the `#relative?` and `#+` of `Addressable::URI` to implement it pretty easily. @zeckalpha, i would merge a PR to that effect if you submitted one.

**Comment by pezra** [comment]
Fix in version 3.13.0
